st2common/st2common/persistence/policy.py

```diff
@@ -26,13 +26,23 @@ def _get_impl(cls):
         return cls.impl
 
     @classmethod
+    def get_by_ref(cls, ref):
+        if ref:
+            ref_obj = PolicyTypeReference.from_string_reference(ref=ref)
+            result = cls.query(name=ref_obj.name, resource_type=ref_obj.resource_type).first()
+            return result
+        else:
+            return None
+
+    @classmethod
     def _get_by_object(cls, object):
-        # PolicyType name is unique.
         name = getattr(object, 'name', '')
-        return cls.get_by_name(name)
+        resource_type = getattr(object, 'resource_type', '')
+        ref = PolicyTypeReference.to_string_reference(resource_type=resource_type, name=name)
+        return cls.get_by_ref(ref)
 
 
-class Policy(Access):
+class Policy(ContentPackResource):
```

Comment:
I settled on resource_type.name. We can definitely revisit this. I'm just focusing on finishing the throttling feature right now.